src/storage-client/src/controller/collection_mgmt.rs

```diff
@@ -288,46 +129,201 @@ impl CollectionManager {
-            id,
-            tx: self.tx.clone(),
-        }
+    pub(super) fn monotonic_appender(
+        &self,
+        id: GlobalId,
+    ) -> Result<MonotonicAppender, StorageError> {
+        let guard = self.collections.lock().expect("CollectionManager panicked");
+        let tx = guard
+            .get(&id)
+            .map(|(tx, _)| tx.clone())
+            .ok_or(StorageError::IdentifierMissing(id))?;
+
+        Ok(MonotonicAppender { tx })
     }
 }
 
+/// Spawns a [`tokio::task`] that will continuously bump the upper for the specified collection,
+/// and append data that is sent via the provided [`mpsc::Sender`].
+///
+/// TODO(parkmycar): One day if we want to customize the tick interval for each collection, that
+/// should be done here.
+/// TODO(parkmycar): Maybe add prometheus metrics for each collection?
+fn write_task<T>(
+    id: GlobalId,
+    write_handle: persist_handles::PersistWriteWorker<T>,
+    now: NowFn,
+) -> (WriteChannel, WriteTask)
+where
+    T: Timestamp + Lattice + Codec64 + From<EpochMillis> + TimestampManipulation,
+{
+    let (tx, mut rx) = mpsc::channel(CHANNEL_CAPACITY);
+
+    let handle = mz_ore::task::spawn(
+        || format!("CollectionManager-write_task-{id}"),
+        async move {
+            let mut interval = tokio::time::interval(DEFAULT_TICK);
+
+            'run: loop {
+                tokio::select! {
+                    // Prefer sending actual updates over just bumping the upper, because sending
+                    // updates also bump the upper.
+                    biased;
+
+                    // Pull as many queued updates off the channel as possible.
+                    cmd = rx.recv_many(CHANNEL_CAPACITY) => {
+                        if let Some(batch) = cmd {
+                            // To rate limit appends to persist we add artifical latency, and will
+                            // finish no sooner than this instant.
+                            let min_time_to_complete = Instant::now() + DEFAULT_TICK;
+
+                            // Reset the interval which is used to periodically bump the uppers
+                            // because the uppers will get bumped with the following update.
+                            interval.reset();
+
+                            let (rows, responders): (Vec<_>, Vec<_>) = batch
+                                .into_iter()
+                                .unzip();
+
+                            // Append updates to persist!
+                            let rows = rows
+                                .into_iter()
+                                .flatten()
+                                .map(|(row, diff)| TimestamplessUpdate { row, diff })
+                                .collect();
+                            let request = vec![(id, rows, T::from(now()))];
+
+                            // We'll try really hard to succeed, but eventually stop.
+                            let retries = Retry::default()
```

Comment:
can you leave a note here that this amount of retrying is almost certainly overkill because we don't expect any contention writing to managed collections? my apprehension is just that this makes it appear as if we are really unsure the amount of effort it's going to take to get the write to succeed, when in reality, we can be pretty confident that the effort is, in fact, very low.

Reply:
Good idea, I added a comment, let me know if you think it's enough